client/db/src/parity_db.rs

```diff
@@ -112,4 +112,8 @@ impl<H: Clone + AsRef<[u8]>> Database<H> for DbAdapter {
 	fn supports_ref_counting(&self) -> bool {
 		true
 	}
+
+	fn sanitize_key<'a>(&self, key: &'a [u8]) -> &'a [u8] {
+		&key[0..key.len() - crate::DB_HASH_LEN]
```

Comment:
```suggestion
		&key[key.len() - crate::DB_HASH_LEN..]
```

Reply:
Thinking about it I would rather keep the existing implementation, drain on vec may be more optimal than reinstantiating the the vec from the slice (I did not look if the compilation did optimize this, just the intention is more clear with drain).